app/models/school_data_importer.rb

```diff
@@ -24,18 +24,47 @@ def schools_data_file
   end
 
   def row_to_school(row)
+    local_authority = LocalAuthority.find_or_initialize_by(code: row.fetch("LA (code)"))
+    local_authority.name = row.fetch("LA (name)")
+    local_authority.save!
+
+    local_authority_district = LocalAuthorityDistrict.find_or_initialize_by(code: row.fetch("DistrictAdministrative (code)"))
+    local_authority_district.name = row.fetch("DistrictAdministrative (name)")
+    local_authority_district.save!
+
+    # Not in public gias extract?
+    # network = Network.find_or_initialize_by(group_uid: row.fetch("Group UID"))
+    # network.group_type = row.fetch("Group Type")
+    # network.group_type_code = row.fetch("Group Type (code)")
+    # network.group_id = row.fetch("Group ID")
+    # network.group_uid = row.fetch("Group UID")
+    # network.name = row.fetch("Group Name")
+
     school = School.find_or_initialize_by(urn: row.fetch("URN"))
+    school.local_authority = local_authority
+    school.local_authority_district = local_authority_district
     school.name = row.fetch("EstablishmentName")
-    school.school_type = row.fetch("TypeOfEstablishment (code)").to_i
+    school.school_type_code = row.fetch("TypeOfEstablishment (code)")
+    school.school_type_name = row.fetch("TypeOfEstablishment (name)")
     school.capacity = row.fetch("SchoolCapacity").to_i
     school.address_line1 = row.fetch("Street")
     school.address_line2 = row.fetch("Locality")
     school.address_line3 = row.fetch("Town")
     school.address_line4 = row.fetch("County (name)")
     school.country = row.fetch("Country (name)")
     school.postcode = row.fetch("Postcode")
+    school.ukprn = row.fetch("UKPRN")
+    school.previous_school_urn = row.fetch("EstablishmentName")
```

Comment:
This doesn't look right, are you sure?

Reply:
Oops, good catch!

Reply:
Aaand I don't see it in the public extract. This is so much fun.

Reply:
`PreviousEstablishmentNumber` I guess

Reply:
Check 

Linked establishments
--

Reply:
(Sorry, whenever I copy-paste from the csv it autoformats as a table 😂 )

Reply:
Linked establishments
--

128816 Predecessor | 128815 Predecessor
-- | --


There seems to be more than one value in that field quite a lot